tests/test_dicom/test_dicom_anonymise.py

```diff
@@ -33,10 +32,9 @@ def test_is_anonymised_dataset():
     ds_anon_func = anonymise_dataset(ds, delete_private_tags=False,
                                      copy_dataset=True)
 
-    # Patient Name should've been the only non-anonymous value. If not,
-    # test data has been changed
-    for elem_manual, elem_func in zip(ds_anon_manual, ds_anon_func):
-        assert elem_manual.value == elem_func.value
+    elems_manual = sorted(list(ds_anon_manual.iterall()), key = lambda x: x.tag)
```

Comment:
`ds.iterall()` skips Sequences, which (along with other things) were causing issues in `ds.__eq__()`

Reply:
```suggestion
    elems_anon_manual = sorted(list(ds_anon_manual.iterall()), key = lambda x: x.tag)
```